src/solver/simulation/timeseries-numbers.cpp

```diff
@@ -682,6 +682,39 @@ bool TimeSeriesNumbers::checkAllElementsIdenticalOrOne(std::vector<uint> w)
     return std::adjacent_find(w.begin(), first_one, std::not_equal_to<uint>()) == first_one;
 }
 
+bool TimeSeriesNumbers::CheckNumberOfColumns(const Study& study)
```

Comment:
Having this method which validates all timeseries with the same logic is a lot better than before I think.

A thought for later though: when adding new timeseries numbers, it will still be quite easy to miss this check. Having a per-object `validate` method that would be called at the start of a computation could make it more "visible", so that we don't miss this validation step. This is linked to the broader validation topic we have already started to discuss.